src/fidesops/ops/api/v1/endpoints/saas_config_endpoints.py

```diff
@@ -245,3 +261,63 @@ def authorize_connection(
         return auth_strategy.get_authorization_url(db, connection_config)
     except FidesopsException as exc:
         raise HTTPException(status_code=HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@router.post(
+    SAAS_CONNECTOR_FROM_TEMPLATE,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONNECTION_INSTANTIATE])],
+    response_model=FidesopsDataset,
+)
+def instantiate_connection_from_template(
+    saas_connector_type: str,
+    template_values: SaasConnectionTemplateValues,
+    db: Session = Depends(deps.get_db),
+) -> FidesopsDataset:
+    """
+    Creates a SaaS Connector and a SaaS Dataset from a template.
+
+    Looks up the connector type in the SaaS connector registry and, if all required
+    fields are provided, persists the associated connection config and dataset to the database.
+    """
+
+    registry: ConnectorRegistry = load_registry(registry_file)
+    connector_template: Optional[ConnectorTemplate] = registry.get_connector_template(
+        saas_connector_type
+    )
+    if not connector_template:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"SaaS connector type '{saas_connector_type}' is not registered.",
+        )
+
+    if DatasetConfig.filter(
+        db=db,
+        conditions=(DatasetConfig.fides_key == template_values.instance_key),
+    ).count():
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"SaaS connector instance key '{template_values.instance_key}' already exists.",
+        )
+
+    try:
+        connection_config: ConnectionConfig = (
+            create_connection_config_from_template_no_save(
+                db, connector_template, template_values
+            )
+        )
+    except KeyOrNameAlreadyExists as exc:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=exc.args[0],
+        )
+
+    connection_config.secrets = validate_secrets(
+        template_values.secrets, connection_config
+    ).dict()
+    connection_config.save(db=db)  # Not persisted to db until secrets are validated
+
+    dataset_config: DatasetConfig = create_dataset_config_from_template(
+        db, connection_config, connector_template, template_values
+    )
+
```

Comment:
The ConnectionConfig needs to be saved to the database before we create a DatasetConfig.  I should be catching all of the known instances that would cause DatasetConfig creation to fail, but it's always possible something happens here. Should I add something that removes the created ConnectionConfig if DatasetConfig creation fails?

Probably.

Reply:
OK I've added this functionality -